### PR TITLE
[SYCL] Implement SYCL 2020 vec aliases

### PR DESCRIPTION
**Pull request by AlexeySachkov**
This commit added aliases for `sycl::vec` class as described in section 4.14.2.2. Aliases of SYCL 2020 specification, revision 6.

Aliases which use OpenCL type, such as `cl_int2` are still preserved, because they are used internally, see intel/llvm#7888.

### CODE REVIEW

**Review by dm-vodopyanov — APPROVED** [review]
Impl LGTM

**Comment by AlexeySachkov** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1477

File: sycl/test/basic_tests/vectors/aliases.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#define CHECK_ALIASES_FOR_VEC_LENGTH(N)                                        \
+  CHECK_ALIAS(char, std::int8_t, N)                                            \
+  CHECK_ALIAS(uchar, std::uint8_t, N)                                          \
+  CHECK_ALIAS(short, std::int16_t, N)                                          \
+  CHECK_ALIAS(ushort, std::uint16_t, N)                                        \
+  CHECK_ALIAS(int, std::int32_t, N)                                            \
+  CHECK_ALIAS(uint, std::uint32_t, N)                                          \
+  CHECK_ALIAS(long, std::int64_t, N)                                           \
+  CHECK_ALIAS(ulong, std::uint64_t, N)                                         \
+  CHECK_ALIAS(half, sycl::half, N)                                       \
```
Comment (bader) [inline sycl/test/basic_tests/vectors/aliases.cpp:21]:
```suggestion
  CHECK_ALIAS(half, sycl::half, N)                                             \
```

File: sycl/include/sycl/aliases.hpp [hunk 2]
```diff
@@ -73,6 +90,9 @@ using ulong = unsigned long;
 using longlong = long long;
 using ulonglong = unsigned long long;
 using half = sycl::detail::half_impl::half;
+
+// FIXME: SYCL 2020 spec says that cl_* aliases should reside in sycl::opencl
+// namespace.
```
Comment (bader) [inline sycl/include/sycl/aliases.hpp:95]: What is blocking us from applying these FIXME comments right away?

Comment (AlexeySachkov) [reply sycl/include/sycl/aliases.hpp:95]: Apparently, we still use some of these aliases: #7888. I don't know exactly about `cl_*` scalar aliases, but we for sure use `cl_*` vector aliases.

I would prefer an incremental approach and get this merged first as a fix for vector aliases and then submit a separate patch for scalar aliases

Comment (AlexeySachkov) [reply sycl/include/sycl/aliases.hpp:95]: I took another look at this: vector `cl_*` aliases depend on scalar `cl_*` aliases and those in turn are still used (#7888)

Comment (bader) [reply sycl/include/sycl/aliases.hpp:95]: > Apparently, we still use some of these aliases: #7888. I don't know exactly about `cl_*` scalar aliases, but we for sure use `cl_*` vector aliases.
> 
> I would prefer an incremental approach and get this merged first as a fix for vector aliases and then submit a separate patch for scalar aliases

I also prefer incremental approach, which should be remove `cl_*` alias usages from DPC++ code base first and remove alias definitions after that. This way there will be no technical debt.

**Comment by AlexeySachkov** [comment]
Post-commit failures are being addressed in #7896

**Comment by AlexeySachkov** [comment]
I've just realized that some of the failures are SYCL-CTS and with the current spec wording, that is a bug in CTS and not in the implementation. Submitted KhronosGroup/SYCL-CTS#446. We should probably temporarily disable some CTS tests to have green post-commit results

**Pull request merged**